src/sentry/workflow_engine/models/alertrule_detector.py

```diff
@@ -0,0 +1,42 @@
+from django.db.models import CheckConstraint, Q, UniqueConstraint
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import DefaultFieldsModel, FlexibleForeignKey, region_silo_model
+
+
+@region_silo_model
+class AlertRuleDetector(DefaultFieldsModel):
+    """
+    A lookup model for rules and detectors.
+    """
+
+    __relocation_scope__ = RelocationScope.Excluded
+
+    alert_rule = FlexibleForeignKey("sentry.AlertRule", null=True)
+    rule = FlexibleForeignKey("sentry.Rule", null=True)
+    detector = FlexibleForeignKey("workflow_engine.Detector")
+
+    class Meta:
+        db_table = "workflow_engine_alertruledetector"
+        app_label = "workflow_engine"
+        unique_together = (
+            ("detector", "rule"),
+            ("detector", "alert_rule"),
+        )
+        constraints = [
+            CheckConstraint(
+                condition=Q(rule__isnull=False, alert_rule__isnull=True)
+                | Q(rule__isnull=True, alert_rule__isnull=False),
+                name="rule_or_alert_rule_detector",
+            ),
+            UniqueConstraint(
+                fields=["rule"],
+                condition=Q(detector__isnull=True),
+                name="unique_rule_user_detector",
+            ),
+            UniqueConstraint(
+                fields=["alert_rule"],
+                condition=Q(detector__isnull=True),
+                name="unique_alert_rule_user_detector",
+            ),
```

Comment:
Both of `UniqueConstraint` have condition that will never be true - `detector` is a `FlexibleForeignKey` which is not allowed to be null by default

Reply:
ah thanks for catching that, I've removed these now. I just need to make sure that either rule or alert_rule is entered (along with detector / workflow depending on the model, but that's already guaranteed)